ValetTests/ValetTests.swift

```diff
@@ -0,0 +1,742 @@
+        XCTAssertEqual(differingAccessibility.allKeys(), Set())
+
+        // Different Class
+        XCTAssertEqual(subclassValet.allKeys(), Set())
+    }
+
+    // MARK: stringForKey / setStringForKey
+
+    func test_stringForKey_isNilForInvalidKey()
+    {
+        XCTAssertNil(valet.stringForKey(key))
+    }
+
+    func test_stringForKey_retrievesStringForValidKey()
+    {
+        XCTAssert(valet.setString(passcode, forKey: key))
+        XCTAssertEqual(passcode, valet.stringForKey(key))
+    }
+
+    func test_stringForKey_equivalentValetsCanAccessSameData()
+    {
+        let equalValet = VALValet(identifier: valet.identifier, accessibility: valet.accessibility)!
+        XCTAssertEqual(0, equalValet.allKeys().count)
+        XCTAssertEqual(valet, equalValet)
+        XCTAssert(valet.setString("monster", forKey: "cookie"))
+        XCTAssertEqual("monster", equalValet.stringForKey("cookie"))
+    }
+
+    func test_stringForKey_withDifferingIdentifier_isNil()
+    {
+        XCTAssert(valet.setString(passcode, forKey: key))
+
+        let differingIdentifier = VALValet(identifier: "wat", accessibility: valet.accessibility)!
+        XCTAssertNil(differingIdentifier.stringForKey(key))
+    }
+
+    func test_stringForKey_withDifferingAccessibility_isNil()
+    {
+        XCTAssert(valet.setString(passcode, forKey: key))
+
+        let differingAccessibility = VALValet(identifier: valet.identifier, accessibility: .AfterFirstUnlockThisDeviceOnly)!
+        XCTAssertNil(differingAccessibility.stringForKey(key))
+    }
+
+    func test_stringForKey_withEquivalentConfigurationButDifferingSubclass_isNil()
+    {
+        XCTAssert(valet.setString("monster", forKey: "cookie"))
+        XCTAssertNil(subclassValet.stringForKey("cookie"))
+    }
+
+    func test_setStringForKey_successfullyUpdatesExistingKey()
+    {
+        XCTAssertNil(valet.stringForKey(key))
+        valet.setString("1", forKey: key)
+        XCTAssertEqual("1", valet.stringForKey(key))
+        valet.setString("2", forKey: key)
+        XCTAssertEqual("2", valet.stringForKey(key))
+    }
+
+    func disabled_test_setStringForKey_failsWithInvalidArguments()
+    {
+        // TODO: see if nilValue tap dance tests are even doable in Swift.
+        var nilVar: String?
+        nilVar = nil
+        XCTAssertFalse(valet.setString(nilVar!, forKey: key))
+    }
+
+    // MARK: Concurrency
+
+    func test_concurrentSetAndRemoveOperations()
+    {
+        let setQueue = dispatch_queue_create("Set String Queue", DISPATCH_QUEUE_CONCURRENT)
+        let removeQueue = dispatch_queue_create("Remove Object Queue", DISPATCH_QUEUE_CONCURRENT)
+
+        for _ in 1...50 {
+            dispatch_async(setQueue, { XCTAssert(self.valet.setString(self.passcode, forKey: self.key)) })
+            dispatch_async(removeQueue, { XCTAssert(self.valet.removeObjectForKey(self.key)) })
+        }
+
+        let setQueueExpectation = self.expectationWithDescription("Set String Queue")
```

Comment:
For clarity when this fails, might want to call this "\(#function) Set String Queue".